js/main.js

```diff
@@ -17,21 +17,46 @@ function formStore(event) {
     notes: $form.elements.notes.value,
     entryId: data.nextEntryId
   };
-  data.nextEntryId++;
-  data.entries.unshift(entry);
-  $imgSrc.setAttribute('src', 'images/placeholder-image-square.jpg');
+  if (data.editing === null) {
+    data.nextEntryId++;
+    data.entries.unshift(entry);
+    $imgSrc.setAttribute('src', 'images/placeholder-image-square.jpg');
 
-  $ul.prepend(renderEntry(entry));
+    $ul.prepend(renderEntry(entry));
+    return;
```

Comment:
We don't want to `return` here, otherwise your `viewSwap`, `toggleNoEntries` and `reset()` wouldn't run